types/layer.go

```diff
@@ -37,6 +37,7 @@ type Layer struct {
 	Binds              interface{}       `yaml:"binds"`
 	Apply              []string          `yaml:"apply"`
 	RuntimeUser        string            `yaml:"runtime_user"`
+	Name               string            // the name of this layer in the stacker file.
```

Comment:
Can we avoid doing this?

Reply:
well right now, given:

```yaml
foo:
  binds:
  apply:
  run: ...
```

We do not have a reference in the `Layer` to `foo` at all.
we could add a reference to the StackerConfig that this layer came from, but that seems more recursive than this.

Reply:
Why not just pass it as an argument to BuildEnvironment(), then?

Reply:
it seems sane to me that a Layer has a name that was tied to it at yaml read time.
Do you have a reason not to?

Reply:
Yes, you can then cache layer builds that were renamed; one of our products builds relied on this at one point, not sure if it still does or not.